pkg/kubelet/server/server.go

```diff
@@ -130,6 +135,8 @@ type filteringContainer struct {
 }
 
 func (a *filteringContainer) Handle(path string, handler http.Handler) {
+	// no tracer provider then noop
+	handler = traces.WithTracing(handler, otel.GetTracerProvider(), "kubelet-container")
```

Comment:
even though it is a no-op, we should still feature gate this.  We should also try and plumb the tracerprovider through to here, rather than use the global one.

Reply:
embedded TracerProvider in filteringContainer struct